src/main/java/com/iscte/engsoft/grupob/CalendarApp/controller/BackendController.java

```diff
@@ -1,17 +1,56 @@
 package com.iscte.engsoft.grupob.CalendarApp.controller;
 
+import com.iscte.engsoft.grupob.CalendarApp.model.UrlFile;
 import org.springframework.http.MediaType;
 import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.net.URL;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 @RestController
 @Validated
 @RequestMapping(value = "/", produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
 public class BackendController {
 
+    private static final Logger LOGGER = Logger.getLogger(BackendController.class.getName());
+
     @RequestMapping(path = "/test", consumes = MediaType.ALL_VALUE)
     public String test() {
         return "Wonderful day to be alive.";
     }
+
+    /**
+     * Returns the json or csv file from the specified URL
```

Comment:
Isto é a descrição do método, portanto: Receives a remote location (url) that contains calendar data, downloads and stores it.